### DISCUSSION TITLE
Support for creating a BaseClient that doesn't extract cookies.

### DISCUSSION THREAD

**Discussion by jonathanslenders** [encode/httpx#2271] [Ideas]
It would be good to be able to create a client that doesn't extract cookies. That way, we can still take advantage of the connection pool, but don't have session information leak from one request to the next.

The implementation for this is probably fairly straightforward. I propose to accept a boolean ``extract_cookies`` in ``BaseClient``/``Client``/``AsyncClient`` which defaults to ``True`` and check for this where we call ``self.cookies.extract_cookies(response)``.

Would you be willing to accept a patch for this?

**Comment by lovelydinosaur** [comment]
Adding `persist_cookies: bool = True` or `extract_cookies: bool = True` both seem like reasonable new bits of API to me, yup.
Alternately we could support `Client(cookies = False)`.

A workaround for the moment would be something like...

```python
class DontPersistCookies(cookies.CookieJar):
    def extract_cookies(self, request, response):
        pass

client = httpx.Client(cookies=DontPersistCookies)
```

(I've not tried it, but should be something like that?)

**Reply by matiuszka** [reply]
@tomchristie 

I think it will not work. The `DontPersistCookies` object will be converted again to the `Cookies` object as per line: https://github.com/encode/httpx/blob/master/httpx/_client.py#L180

But this will work, but it is very ugly:
```python
class DontPersistCookies(cookies.CookieJar):
    def extract_cookies(self, request, response):
        pass

client = httpx.Client()
client._cookies = DontPersistCookies()
```

I will create a PR with the fixes for this, probably I will go with `extract_cookies: bool = True`.

**Reply by lovelydinosaur** [reply]
> The DontPersistCookies object will be converted again to the Cookies object

It'll be used as the underlying cookie jar within the `Cookies` instance.
Which oughta work as expected.

**Reply by matiuszka** [reply]
Oh, you are right. I thought that you did inheritance from `Cookies`. Thanks for this.

**Comment by jonathanslenders** [comment]
Thank you @tomchristie ! That works. If we find some time, any of us will contribute a pull request.
